pin-helper.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+
+PIN_COMMAND='
+set -e
+pip install -q -U pip pip-compile-multi
+pip-compile-multi -u \
+    -g base -g test -g dev -g docs
+'
+
+PIN_PYTHON=3.7
+docker run -t -v $PWD:/src -w /src python:$PIN_PYTHON bash -c "${PIN_COMMAND}"
+
+set +x
+echo "###########################################################"
+echo "# Running tests on different python versions              #"
+echo "# Pinned packages already saved. You may skip with CTRL+C #"
+echo "###########################################################"
+set -x
+
+# Don't quit if a version fails
+set +e
```

Comment:
I guess the loop is to highlight failures in case the generated dependencies don't work on a later version of Python. If so, what's the use case for letting these commands silently fail?

Reply:
Improved it now